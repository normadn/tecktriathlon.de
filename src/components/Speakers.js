import React from "react";
import SpeakerCard from "./SpeakerCard";
import SpeakerSlider from "./SpeakerSlider";
import Mentors from "./Mentors";

import cubes3webp from "../assets/img/icons/cubes-3.webp";
import cubes3png from "../assets/img/icons/cubes-3.png";
import norman from "../assets/img/speakers/norman1.jpg";
import andrei from "../assets/img/speakers/andrei.jpg";
import sina from "../assets/img/speakers/sina2.jpg";

import ContentWrapper from "./ContentWrapper";

const speakers = [
  {
    name: "Norman Saade",
    title: "Software-Entwickler",
    description:
      "Norman's Spezialgebiete sind Mobile Apps, Websiten und Datenanalyse. Er ist studierte Wirtschaftsingenieur (KIT, Master of Science) und hat bereits mehrere eigene Unternehmen gegründet und auch für Firmen wie z.B. Google gearbeitet, wo er auch Andrei kennenlernte. In seiner Freizeit liebt er es viel Zeit in der Natur zu verbringen und sich sportlich zu fordern.",
    image: norman
  },
  {
    name: "Andrei Duma",
    title: "Software-Entwickler",
    description:
      "Andrei ist Vollblut Software Entwickler. Am Liebsten konzentriert er sich auf Backend-Anwendungen, aber auch im Frontend ist er zu Hause. Er hat Informatik studiert und nebenbei vier Praktika bei Google absolviert bevor er für mehrere Jahre für Adobe arbeitete. Andrei ist passionierter Bergsteiger und reist in seiner Freizeit gerne um die Welt.",
    image: andrei
  },
  {
    name: "Sina Eitel",
    title: "Web-Designerin",
    description:
      "Sina designt in ihrer Freizeit gerne Logos, Flyer und Web-Anwendungen, aber auch sie kann im Frontend Hand anlegen. Sie ist das Organisationtalent im Team und sorgt mit ihrem Lachen stets für gute Stimmung. Beruflich ist Sina als Polizistin unterwegs und bringt hier Verbrecher zur Strecke.",
    image: sina
  }
];

const Speakers = () => (
  <section
    css={{
      backgroundColor: "#f8f8f8",
      position: "relative"
    }}
    id="team"
  >
    <ContentWrapper>
      <h1
        css={{
          paddingTop: "112px",
          paddingBottom: "48px",
          textAlign: "center",
          position: "relative",
          zIndex: "3",
          "@media(max-width: 820px)": {
            paddingTop: "100px",
            paddingBottom: "24px"
          }
        }}
      >
        amsel.tech Team
      </h1>
      {/* Desktop */}
      <div
        css={{
          display: "grid",
          position: "relative",
          zIndex: 3,
          maxWidth: speakers.length * 500,
          marginLeft: "auto",
          marginRight: "auto",
          gridTemplateColumns: speakers
            .map(() => "1fr ")
            .reduce((a, b) => a + b),
          "@media(max-width: 1120px)": {
            display: "none"
          }
        }}
      >
        {speakers.map((speaker) => (
          <SpeakerCard {...speaker} key={speaker.name} />
        ))}
      </div>
      {/* Mobile */}
      <div
        css={{
          paddingBottom: "30px",
          "@media(min-width: 1120px)": { display: "none" }
        }}
      >
        <SpeakerSlider speakers={speakers} />
      </div>
      <picture
        css={{
          zIndex: "1",
          width: "235px",
          height: "289px",
          position: "absolute",
          bottom: "-250px",
          right: "62px",
          display: "block",
          "@media(max-width: 1000px)": {
            display: "none"
          }
        }}
      >
        <source srcSet={cubes3webp} type="image/webp" />
        <source srcSet={cubes3png} type="image/png" />
        <img
          css={{
            "@media(max-width: 1000px)": {
              display: "none"
            }
          }}
          src={cubes3png}
          alt="Floating cubes"
        />
      </picture>
      {/*<Mentors /> */}
    </ContentWrapper>
  </section>
);

export default Speakers;
