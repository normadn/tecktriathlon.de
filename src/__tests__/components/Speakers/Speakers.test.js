import React from "react";
import Speakers from "../../../components/Speakers";
import { shallow } from "enzyme";

describe("Speakers", () => {
  it("renders correctly", () => {
    expect(shallow(<Speakers />)).toMatchSnapshot();
  });
});
