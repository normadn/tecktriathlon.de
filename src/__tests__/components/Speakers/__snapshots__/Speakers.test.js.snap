// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speakers renders correctly 1`] = `
<section
  css={
    Object {
      "backgroundColor": "#f8f8f8",
      "position": "relative",
    }
  }
  id="speakers"
>
  <_default>
    <h1
      css={
        Object {
          "@media(max-width: 820px)": Object {
            "paddingBottom": "24px",
            "paddingTop": "200px",
          },
          "paddingBottom": "48px",
          "paddingTop": "112px",
          "position": "relative",
          "textAlign": "center",
          "zIndex": "3",
        }
      }
    >
      Fireside Chat Speakers
    </h1>
    <div
      css={
        Object {
          "@media(max-width: 1120px)": Object {
            "display": "none",
          },
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr 1fr ",
          "marginLeft": "auto",
          "marginRight": "auto",
          "maxWidth": 1500,
          "position": "relative",
          "zIndex": 3,
        }
      }
    >
      <SpeakerCard
        description="David is the Co-Founder and current Head of Design at Ada. He graduated from Queen’s University’s Geological and Earth Sciences program in 2012. At Ada, he is designing and built tools that allow companies to answer common questions automatically, improving customer satisfaction and saving everyone's time. Ada is used by Telus, Medium, Wattpad, Shopify and millions of their customers."
        image="test-file-stub"
        key="David Hariri"
        name="David Hariri"
        title="Co-Founder of Ada"
      />
      <SpeakerCard
        description="Kaheer is a Principal Research Program Manager at the Microsoft Research Montreal lab.  Kaheer co-founded the deep learning for language startup Maluuba and served as its CTO prior to its acquisition, by Microsoft,  in early 2017.  He currently works on machine learning approaches for natural language processing focusing on question answering, conversation systems and common sense reasoning."
        image="test-file-stub"
        key="Kaheer Suleman"
        name="Kaheer Suleman"
        title="Principal Research Program Manager at Microsoft"
      />
      <SpeakerCard
        description="Sarah is the Founder and CEO of Valkerie, a boutique tech growth consultancy based in Toronto ON. She’s spent her career working on early stage technology startups, starting as a foundational team member of Wave, a Canadian financial technology giant. More recently she led the International Community Growth team at the San Francisco payments startup, Tilt, which was acquired by Airbnb in 2017."
        image="test-file-stub"
        key="Sarah Stockdale"
        name="Sarah Stockdale"
        title="Founder & CEO of Valkerie"
      />
    </div>
    <div
      css={
        Object {
          "@media(min-width: 1120px)": Object {
            "display": "none",
          },
          "paddingBottom": "30px",
        }
      }
    >
      <SpeakerSlider
        speakers={
          Array [
            Object {
              "description": "David is the Co-Founder and current Head of Design at Ada. He graduated from Queen’s University’s Geological and Earth Sciences program in 2012. At Ada, he is designing and built tools that allow companies to answer common questions automatically, improving customer satisfaction and saving everyone's time. Ada is used by Telus, Medium, Wattpad, Shopify and millions of their customers.",
              "image": "test-file-stub",
              "name": "David Hariri",
              "title": "Co-Founder of Ada",
            },
            Object {
              "description": "Kaheer is a Principal Research Program Manager at the Microsoft Research Montreal lab.  Kaheer co-founded the deep learning for language startup Maluuba and served as its CTO prior to its acquisition, by Microsoft,  in early 2017.  He currently works on machine learning approaches for natural language processing focusing on question answering, conversation systems and common sense reasoning.",
              "image": "test-file-stub",
              "name": "Kaheer Suleman",
              "title": "Principal Research Program Manager at Microsoft",
            },
            Object {
              "description": "Sarah is the Founder and CEO of Valkerie, a boutique tech growth consultancy based in Toronto ON. She’s spent her career working on early stage technology startups, starting as a foundational team member of Wave, a Canadian financial technology giant. More recently she led the International Community Growth team at the San Francisco payments startup, Tilt, which was acquired by Airbnb in 2017.",
              "image": "test-file-stub",
              "name": "Sarah Stockdale",
              "title": "Founder & CEO of Valkerie",
            },
          ]
        }
      />
    </div>
    <picture
      css={
        Object {
          "@media(max-width: 1000px)": Object {
            "display": "none",
          },
          "bottom": "-250px",
          "display": "block",
          "height": "289px",
          "position": "absolute",
          "right": "62px",
          "width": "235px",
          "zIndex": "1",
        }
      }
    >
      <source
        srcSet="test-file-stub"
        type="image/webp"
      />
      <source
        srcSet="test-file-stub"
        type="image/png"
      />
      <img
        alt="Floating cubes"
        css={
          Object {
            "@media(max-width: 1000px)": Object {
              "display": "none",
            },
          }
        }
        src="test-file-stub"
      />
    </picture>
    <_default />
  </_default>
</section>
`;
